python/kfserving/kfserving/handlers/http.py

```diff
@@ -49,14 +49,21 @@ def post(self, name: str):
         request = self.validate(request)
         response = model.predict(request)
         response = model.postprocess(response)
-        self.write(json.dumps(response))
+        self.write(response)
 
 
 class ExplainHandler(HTTPHandler):
     def post(self, name: str):
         model = self.get_model(name)
-        request = self.validate(self.request)
+        try:
+            body = json.loads(self.request.body)
+        except json.decoder.JSONDecodeError as e:
+            raise tornado.web.HTTPError(
+                status_code=HTTPStatus.BAD_REQUEST,
+                reason="Unrecognized request format: %s" % e
+            )
+        request = self.validate(body)
```

Comment:
should `validate` also be called after `preprocess` like in predict ? The request may come in with arbitrary data plane but after preprocess it should conform to KFServing data plane for prediction.

Reply:
Fixed it.